src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.SpecialTypes.cs

```diff
@@ -154,56 +154,64 @@ public static object[][] DefaultValues
-                new object[] {"ulong", "ulong.MinValue", ulong.MinValue, true },
-                new object[] {"ushort", "ushort.MaxValue", ushort.MaxValue, true },
-                new object[] {"ushort", "ushort.MinValue", ushort.MinValue, true },
-            };
+            return
+            [
+                ["string?", "default", default(string), true],
+                ["string", "\"test\"", "test", true],
+                ["string", "\"a\" + \"b\"", "ab", true],
+                ["DateOnly?", "default", default(DateOnly?), false],
+                ["bool", "default", default(bool), true],
+                ["bool", "false", false, true],
+                ["bool", "true", true, true],
+                ["System.Threading.CancellationToken", "default", default(CancellationToken), false],
+                ["Todo?", "default", default(Todo), false],
+                ["char", "\'a\'", 'a', true],
+                ["int", "default", 0, true],
+                ["int", "1234", 1234, true],
+                ["int", "1234 * 4", 1234 * 4, true],
+                ["double", "1.0", 1.0, true],
+                ["double", "double.NaN", double.NaN, true],
+                ["double", "double.PositiveInfinity", double.PositiveInfinity, true],
+                ["double", "double.NegativeInfinity", double.NegativeInfinity, true],
+                ["double", "double.E", double.E, true],
+                ["double", "double.Epsilon", double.Epsilon, true],
+                ["double", "double.NegativeZero", double.NegativeZero, true],
+                ["double", "double.MaxValue", double.MaxValue, true],
+                ["double", "double.MinValue", double.MinValue, true],
+                ["double", "double.Pi", double.Pi, true],
+                ["double", "double.Tau", double.Tau, true],
+                ["float", "float.NaN", float.NaN, true],
+                ["float", "float.PositiveInfinity", float.PositiveInfinity, true],
+                ["float", "float.NegativeInfinity", float.NegativeInfinity, true],
+                ["float", "float.E", float.E, true],
+                ["float", "float.Epsilon", float.Epsilon, true],
+                ["float", "float.NegativeZero", float.NegativeZero, true],
+                ["float", "float.MaxValue", float.MaxValue, true],
+                ["float", "float.MinValue", float.MinValue, true],
+                ["float", "float.Pi", float.Pi, true],
+                ["float", "float.Tau", float.Tau, true],
+                ["decimal", "decimal.MaxValue", decimal.MaxValue, true],
+                ["decimal", "decimal.MinusOne", decimal.MinusOne, true],
+                ["decimal", "decimal.MinValue", decimal.MinValue, true],
+                ["decimal", "decimal.One", decimal.One, true],
+                ["decimal", "decimal.Zero", decimal.Zero, true],
+                ["long", "long.MaxValue", long.MaxValue, true],
+                ["long", "long.MinValue", long.MinValue, true],
```

Comment:
```suggestion
                ["long", "long.MinValue", long.MinValue, true],
                ["long", "3.14", 3.14, true],
```
Non-enum types don't match